astra/src/test/java/com/slack/astra/chunkManager/CachingChunkManagerTest.java

```diff
@@ -160,6 +168,81 @@ public void testAddMessageIsUnsupported() throws TimeoutException {
         .isInstanceOf(UnsupportedOperationException.class);
   }
 
+  @Test
+  public void testCreatesChunksOnAssignment() throws Exception {
+    System.setProperty("astra.ng.dynamicChunkSizes", "true");
+
+    cachingChunkManager = initChunkManager();
+    initAssignments();
+
+    await()
+        .timeout(10000, TimeUnit.MILLISECONDS)
+        .until(() -> cachingChunkManager.getChunksMap().size() == 1);
+    assertThat(cachingChunkManager.getChunksMap().size()).isEqualTo(1);
+
+    // assert state is correct
+    // assert searchMetadata is correct
+    // assert can query
+    System.setProperty("astra.ng.dynamicChunkSizes", "false");
```

Comment:
I like the resetting at the end - consider trying to reset it back to the previous value. It would be safer to read the current value at the beginning, and then reset it to whatever that was at the end (true, false, or empty/deleted).